### DISCUSSION TITLE
Convert experience.dataset to torch.utils.data.Dataset

### DISCUSSION THREAD

**Discussion by SinHanYang** [ContinualAI/avalanche#1108] [Q&A] [answered]
Hi, are there any ways to convert `<class 'avalanche.benchmarks.utils.avalanche_dataset.AvalancheSubset'>` to `torch.utils.data.Dataset `?

I'm using `train_stream` for CL tasks. The dataset for each task are as follows:
```
for experience in train_stream:
    training_dataset = experience.dataset
    print(type(training_dataset))
```
It shows `<class 'avalanche.benchmarks.utils.avalanche_dataset.AvalancheSubset'>`.

Since I need to save this dataset for analysis, the first step is converting dataset to PyTorch's dataset. However, I didn't find a way to do so.

Thanks

**Comment by AntonioCarta** [comment, marked as answer]
You don't need to convert it since `AvalancheSubset` implements the same methods of pytorch datasets. However, keep in mind that it's not optimized to be serialized (this is something that we are working on).

Do you want to save the stream dataset or the replay buffer? if you want the former, I would suggest to avoid serialization and instead create a new benchmark with the same arguments by setting a fixed seed.

**Reply by SinHanYang** [reply]
Thanks for reply !

I found that pytorch DataLoader can load `AvalancheSubset` since it's an inheritance of pytorch dataset.
I will save `AvalancheSubset` to a file by `pickle` and load it after for another experiment.
